torch/fx/experimental/symbolic_shapes.py

```diff
@@ -111,16 +111,24 @@ class PySymInt(object):
     implementation of symbolic shapes.
     """
     def __init__(self, expr, shape_env, constant=None):
-        self.expr = expr
+        self._expr = expr
         self.shape_env = shape_env
         self.constant = constant
 
+    @property
+    def expr(self):
+        self._update_expr()
+        return self._expr
+
     def wrap(self, num):
         return PySymInt(sympy.Integer(num), self.shape_env, constant=num)
 
     def clone(self):
         return PySymInt(self.expr, self.shape_env, constant=self.constant)
 
+    def _update_expr(self):
+        self._expr = self.shape_env.simplify(self._expr)
```

Comment:
do you think this will noticeably slow down compilation times 🤔. I hope `simplify()` is a fast no-op

Reply:
We have a bunch of caches, so it'll never get run twice.

Reply:
Not sure about that? We'd have to audit all expr call sites no?

Reply:
What do you mean? I'm saying that `self.shape_env.simplify` sits behind a LRU cache, so if we hit it twice it should mostly be a no-op.